permissions/resource_rule_set.go

```diff
@@ -0,0 +1,177 @@
+package permissions
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/databricks/databricks-sdk-go/service/iam"
+	"github.com/databricks/terraform-provider-databricks/common"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+type RuleSetApi struct {
+	client *common.DatabricksClient
+	ctx    context.Context
+}
+
+func NewRuleSetApi(ctx context.Context, m interface{}) RuleSetApi {
+	return RuleSetApi{m.(*common.DatabricksClient), ctx}
+}
+
+func (a RuleSetApi) Read(getRuleSetReq iam.GetRuleSetRequest) (*iam.RuleSetResponse, error) {
```

Comment:
try changing the signature to:
`func ReadFromWsOrAcc(ctx context.Context, client *common.DatabricksClient, req ) getRuleSetReq iam.GetRuleSetRequest) (*iam.RuleSetResponse, error)`

Reply:
and make this function confined into `ResourceRuleSet`.